components/esp_modem/examples/modem_console/main/modem_console_main.cpp

```diff
@@ -59,8 +64,37 @@ using namespace esp_modem;
 static SignalGroup exit_signal;
 
 
+extern "C" void config_gpio(void)
```

Comment:
No need for the C linkage
```suggestion
void config_gpio(void)
```